src/Plugin/ElasticsearchIndexBase.php

```diff
@@ -129,38 +137,47 @@ public function getIndexDefinition(array $context = []) {
   }
 
   /**
-   * @inheritdoc
+   * {@inheritdoc}
    */
   public function getExistingIndices() {
-    $params = [
-      'index' => $this->indexNamePattern(),
-    ];
-
     try {
-      return array_keys($this->client->indices()->get($params));
+      $params = [
+        'index' => $this->indexNamePattern(),
+      ];
+
+      // Get a list of indices.
+      $operation = ElasticsearchOperations::INDEX_GET;
+      $request_event = new ElasticsearchOperationRequestEvent([$this->client->indices(), 'get'], [$params], $operation, $this);
```

Comment:
Maybe we can have the event operation strings (ie. 'get', 'delete') as constants in ElasticsearchOperationRequestEvent class?

Reply:
The `get` here is the method in the client class. The operation itself is defined as a string in `ElasticsearchOperations::INDEX_GET`.